server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -248,9 +248,8 @@ private void startNewSnapshots(List<SnapshotsInProgress.Entry> snapshotsInProgre
         }
     }
 
-    private void startNewShards(SnapshotsInProgress.Entry entry, Map<ShardId, IndexShardSnapshotStatus> startedShards) {
+    private void startNewShardSnapshots(SnapshotsInProgress.Entry entry, Map<ShardId, IndexShardSnapshotStatus> startedShards) {
```

Comment:
May be `snapshotNewShards`? I believe we use `snapshot` as a verb quiet often. Also all shards belong to the same snapshot?

Reply:
I just tried to keep the names aligned with the existing methods. The caller of this is called `startNewSnapshots` and since each snapshot is broken down into snapshots of the shards (of the indices), I thought `startNewShardSnapshot` might fit better. I kept the `start*` naming since it conveys that the tasks are forked off. Is the naming confusing?

I think the `new` here just means those listed in the snapshotsInProgress, and I think, yes, all shards called here are within the same snapshot.